doc/CMakeLists.txt

```diff
@@ -1,2 +1,122 @@
-add_subdirectory(doxygen)
-add_subdirectory(sphinx)
+# Main documentation targets
+add_custom_target_subproject(doc sequant)
+
+# Find Python for Sphinx dependencies
+find_package(Python 3.8 COMPONENTS Interpreter REQUIRED)
+
+# Find Doxygen for API documentation
+find_package(Doxygen 1.9.3)
```

Comment:
Do we really need v1.9.3 or higher? That is, 1.9.2 doesn't work?
_If_ Doxygen was following semantic versioning, I would expect any 1.9.x to work - hence the question